nixos/modules/services/continuous-integration/gitlab-runner.nix

```diff
@@ -0,0 +1,51 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.gitlab-runner;
+  configFile = pkgs.writeText "config.toml" cfg.configText;
+in
+{
+  options.services.gitlab-runner = {
+    enable = mkEnableOption "Gitlab Runner";
+
+    configText = mkOption {
+      description = "Verbatim config.toml to use";
```

Comment:
Recently I learned this alternative from roblabla to generate toml configuration from native nix expressions: 

https://github.com/Mic92/nixpkgs/blob/8fddcad3f9831d1961ffd8f6f35f14c759a99cdd/nixos/modules/services/monitoring/telegraf.nix#L11

If found it more readable, but your mileage may vary.